docs/architecture/deep-dives/passkeys/implementations/provider/browser-extension.md

```diff
@@ -0,0 +1,99 @@
+# Browser Extension
+
+Bitwarden supports passkeys in all browser by implementing the WebAuthn API. The browser does not
+need to support WebAuthn natively, the extension will polyfill if necessary, effectively adding
+support for passkeys to any browser.
+
+## Compatibility
+
+There are currently no browser APIs that allow extensions to provide passkeys alongside the
+browser's native implementation. This means that the only way an extension can provide passkeys is
+by completely replacing the native implementation with its own. This is done by injecting a script
+into the page that replaces the native implementation with the extension's implementation. In
+practice this is done by reassigning the `navigator.credentials.create` and
+`navigator.credentials.get` methods.
+
+To avoid implementing support for the entire FIDO2 ecosystem (hardware keys, CaBLE, etc.), the
+extension retains the ability to trigger the native implementation (usually referred to as a
+"fallback"), if the user chooses to not proceed with Bitwarden. In practice this is done by storing
+references to the native functions in separate variables before reassigning.
+
+## Architecture
+
+Bitwarden implements a simplified version of the FIDO2 architecture based solely on the WebAuthn API
+specification. This is because the embedded `FIDO2 Authenticator` will never be used in a standalone
```

Comment:
Nit: Provide link to spec
```suggestion
Bitwarden implements a simplified version of the FIDO2 architecture based solely on the [WebAuthn API
specification](https://www.w3.org/TR/webauthn-3/). This is because the embedded `FIDO2 Authenticator` will never be used in a standalone
```